packages/firebase_messaging/android/src/main/java/io/flutter/plugins/firebasemessaging/FirebaseMessagingPlugin.java

```diff
@@ -99,7 +110,28 @@ public void onReceive(Context context, Intent intent) {
 
   @Override
   public void onMethodCall(final MethodCall call, final Result result) {
-    if ("configure".equals(call.method)) {
+    if ("FcmDartService.start".equals(call.method)) {
+      long setupCallbackHandle = 0;
+      long backgroundMessageHandle = 0;
+      try {
+        List<Long> callbacks = ((ArrayList<Long>) call.arguments);
+        setupCallbackHandle = callbacks.get(0);
+        backgroundMessageHandle = callbacks.get(1);
+      } catch (Exception e) {
+        Log.e(TAG, "There was an exception when getting callback handle from dart side");
+        e.printStackTrace();
+      }
+      FlutterFirebaseMessagingService.setBackgroundSetupHandle(
```

Comment:
I'm concerned about all these static calls. Ideally we wouldn't have any static access. @collinjackson it would be great if we could find a better solution for this in both this plugin and the android_alarm_manager. Maybe there is an opportunity with the new embedding...

In fact, @kroikie do you know what happens if this plugin is registered with 2 different isolates?

Reply:
@matthew-carroll If the new embedding is a better solution I'd be happy to transition to it, I am not sure it is though could you point me.

FWIU the goal is to have the plugin registered in both isolates with each isolate having its own channel for message passing.

Reply:
I suppose we probably can't use the new embedding until we call it stable later in the year, so don't worry about that.